### ISSUE TITLE
Problem with ControlStructureSpacing.LineAfterClose ("if" inside switch case)

### ISSUE THREAD

**Issue by glen-84** [squizlabs/PHP_CodeSniffer#921]
Code example:

```php
        if (1) {
            $x = 0;
        }

        switch (true) {
            case 1:
                if (2) {
                    $x = 0;
                }

                break;
            case 2:
                $x = 0;

                break;
        }
```

The second `if` statement gives a "Blank line found after control structure" – is this a bug, or why is it disallowed?

**Comment by aik099** [comment]
It's a regression bug, yes. It was fixed in `master` branch, but wasn't released yet.

**Comment by glen-84** [comment]
Thanks @aik099.

**Comment by gsherwood** [comment]
I actually can't replicate any problems on the latest stable version either. That code sample doesn't produce any errors in any of the included standards. Do you have a piece of code and a command I can use to replicate the problem and confirm it is fixed?

**Comment by glen-84** [comment]
```php
<?php

namespace X;

class Test
{
    public function test()
    {
        switch (1) {
            case 2:
                if (3) {
                    $x = 4;
                }

                break;
        }
    }
}
```

```
 13 | ERROR | [x] Blank line found after control structure
    |       |     (Squiz.WhiteSpace.ControlStructureSpacing.LineAfterClose)
```

```
$ phpcs --version
PHP_CodeSniffer version 2.5.1 (stable) by Squiz (http://www.squiz.net)
```

**Comment by gsherwood** [comment]
Thanks. So what exactly is the problem you are reporting then? The sniff returning the error is part of the Squiz standard and it doesn't want a newline after the IF because it is followed by another closing token (the break in this case).

If you run phpcbf over it, you can see it actually wants this code:
```php
<?php

namespace X;

class Test
{


    public function test()
    {
        switch (1) {
            case 2:
                if (3) {
                    $x = 4;
                }
            break;
        }

    }//end test()


}//end class
```

**Comment by glen-84** [comment]
That's just weird. I indent other breaks without any errors, like:

```php
<?php

namespace X;

class Test
{
    public function test()
    {
        switch (1) {
            case 2:
                if (3) {
                    $x = 4;
                }

                break;
            case 5:
                $x = 6;

                break; // <- No error here.
        }
    }
}
```

Why does it want the break in line with the case?

Anyway, I'll have to disable this rule.

Thanks for your help.

**Comment by gsherwood** [comment]
> That's just weird. I indent other breaks without any errors, like:

That code produces the same errors.
```
 15 | ERROR   | [x] Case breaking statement must be indented 4 spaces from SWITCH keyword (Squiz.ControlStructures.SwitchDeclaration.BreakIndent)
 19 | ERROR   | [x] Case breaking statement must be indented 4 spaces from SWITCH keyword (Squiz.ControlStructures.SwitchDeclaration.BreakIndent)
```
I don't know what your standard looks like, but if you are including the `Squiz.ControlStructures.SwitchDeclaration` sniff then you will be getting those errors.

> Why does it want the break in line with the case?

Because that's what coding standard we use within Squiz. No other reason than that.

**Comment by glen-84** [comment]
I see. We're not using `Squiz.ControlStructures.SwitchDeclaration`.

**Comment by gsherwood** [comment]
> I see. We're not using Squiz.ControlStructures.SwitchDeclaration.

Then maybe you wont be getting any errors at all for those break statements. The other sniff you've included is giving you an error for the closing IF brace, not the break statement itself.

Just remove the checks you dont want and you should be fine. If you need help figuring out which ones need to go, just let me know.

**Comment by glen-84** [comment]
Thanks Greg. =)
